### ISSUE TITLE
MIC can't get access to Azure Instance Metadata service

### ISSUE THREAD

**Issue by Kuzbekov** [Azure/aad-pod-identity#484]
When we use MSI for kubernetes, MIC can't update VM's because its requests catched by NMI.
Even if I label that pod with correct label it still can't assign Identity to itself.
Temporary killing NMI DaemonSet unlock MIC and allows it to correctly do all assignments.

> [aad-pod-identity-mic-f4c46b576-vc5h6] I0116 07:50:47.348141       1 mic.go:787] Processing node usazuenade00346, add [44], del [44] 
> [aad-pod-identity-mic-f4c46b576-vc5h6] E0116 07:52:07.433303       1 vmss.go:103] azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Compute/virtualMachineScaleSets/REDACTED_scalesetname?api-version=2018-04-01: StatusCode=403 -- Original Error: adal: Refresh request failed. Status Code = '403'. Response body: no AzureAssignedIdentity found for pod:default/aad-pod-identity-mic-f4c46b576-vc5h6 in assigned state 
> [aad-pod-identity-mic-f4c46b576-vc5h6] E0116 07:52:07.433367       1 mic.go:831] Updating msis on node REDACTED_scalesetname, add [44], del [44] failed with error azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Compute/virtualMachineScaleSets/REDACTED_scalesetname?api-version=2018-04-01: StatusCode=403 -- Original Error: adal: Refresh request failed. Status Code = '403'. Response body: no AzureAssignedIdentity found for pod:default/aad-pod-identity-mic-f4c46b576-vc5h6 in assigned state 

**Steps To Reproduce**
Deploy Kubernetes with AKS-engine using  Managed Identity account.
Deploy aad-pod-identity with helm chart from this repository

**Expected behavior**
requests to 169.254.169.254 from mic pod should be passed to Azure Instance Metadata service rather then catched by nmi

**AAD Pod Identity version**
1.5.4

**Kubernetes version**
1.14.7 Deployed with AKS-Engine Using Managed Identity

**Comment by kkmsft** [comment]
Thank you for using aad pod identity @Kuzbekov . 
We are looking into this. There is code in NMI which ensures that request from MIC can directly talk to IMDS in case of managed identity. But the pattern which is checked is following the MIC name from the standalone deployment file (mic-*) only and thus blocks the helm chart deployed(aad-pod-identity-mic-*) MIC. Will raise a PR shortly on this.

**Comment by Kuzbekov** [comment]
Probably it will be more convenient and secure to create special label that will force NMI to passthrough all requests. I can try to do that if and contribute if you need a help.

**Comment by kkmsft** [comment]
@Kuzbekov - There is a way in which an AzurePodIdentityException (https://github.com/Azure/aad-pod-identity/blob/master/docs/readmes/README.app-exception.md) can be created to exclude applications from being intercepted by NMI. 

Since this is available in the latest release, can you please give that a shot to exclude the MIC using the mic labels and provide us feedback on that.

**Comment by elsesiy** [comment]
@kkmsft I was able to get this working with the said exception in place:
```
apiVersion: "aadpodidentity.k8s.io/v1"
kind: AzurePodIdentityException
metadata:
  name: test-exception
spec:
  PodLabels:
    app: mic
    component: mic
```

Is there any reason why the NMI would ever intercept traffic from MIC? Otherwise I agree with @Kuzbekov and the exception should be default.

**Comment by UmairSyed** [comment]
Is this issue fixed in nmi:1.6.0? 
I am facing the same issue as my error from nmi is:
Status (404) took ns" req.method=GET req.path=/metadata/identity/oauth2/token req.remote=
time= level=error msg="no AzureAssignedIdentity found for pod in assigned state, context canceled" req.method=GET req.path=/metadata/identity/oauth2/token req.remote=
but the MIC is showing this error:
Original Error: Code="LinkedAuthorizationFailed" Message="The client 'xxx' with object id 'xxx' has permission to perform action 'Microsoft.Compute/virtualMachineScaleSets/write' on scope '/subscriptions/xxx/resourceGroups/MC_dms-ti-PRD-aue-rg_aks/providers/Microsoft.Compute/virtualMachineScaleSets/aks-nodepool1-xxx-vmss'; however, it does not have permission to perform action 'Microsoft.ManagedIdentity/userAssignedIdentities/assign/action' on the linked scope(s) '/subscriptions/xxx/resourcegroups/PRD-aue-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/dmsti-prd-mi-aks-pod-identity' or the linked scope(s) are invalid."

**Comment by aramase** [comment]
@UmairSyed This issue was fixed with helm chart release - 1.5.6. From the error you posted above, it looks like you are missing the required role assignments.

Please follow the doc here for performing the required role assignments - https://github.com/Azure/aad-pod-identity/blob/master/docs/readmes/README.role-assignment.md
